Add HadoopCatalog example to create/load tables in the Iceberg User docs

hzfanxinxin opened the pull request:
This PR propose to add HadoopCatalog example to create/load tables in the Iceberg User docs.

hzfanxinxin commented on site/docs/api-quickstart.md:
```diff
@@ -48,6 +48,36 @@ logsDF.write
 
+### Using a Hadoop catalog
+
+The Hadoop catalog doesn't need to connects to a Hive MetaStore. To get a Hadoop catalog see:
```
I think so.

hzfanxinxin commented on site/docs/api-quickstart.md:
````diff
@@ -48,6 +48,36 @@ logsDF.write
+val catalog = new HadoopCatalog(conf, warehousePath);
+```
+
+Like Hive catalog, Hadoop catalog implements the interface `Catalog`. So it also contains methods for working with tables, like createTable, loadTable, renameTable, and dropTable.
````
Yes, you are right. So should i remove renameTable?

rdblue replied:
Yes, please.

hzfanxinxin commented on site/docs/api-quickstart.md:
```diff
@@ -48,6 +48,36 @@ logsDF.write
+import org.apache.iceberg.hadoop.HadoopCatalog;
+
+val conf = new Configuration();
+val warehousePath = "hdfs://warehouse_path";
```
`hdfs://warehouse/path`, is it ok?

rdblue replied:
No, then `warehouse` is the authority section. You should change it to something like `hdfs://host:8020/warehouse/path/`

hzfanxinxin commented on site/docs/api-quickstart.md:
```diff
@@ -48,6 +48,36 @@ logsDF.write
+logsDF.write
+    .format("iceberg")
+    .mode("append")
+    .save("hdfs://warehouse_path/logging/logs")
```
Yes, this is also a problem that bothers me. I look the method `findTable` in class IcebergSource.java, it just support `HadoopTables` and `HiveCatalog`. So i think if we can add a option like `.option("catalog","HadoopCatalog")` to the DataSourceOptions, and the default value of catalog option is HiveCatalog.  what's your opinion on this?

rdblue replied:
This seems like something we might want to solve in configuration instead of hard-coding support for `HiveCatalog`. This problem might go away when we get a better API in Spark 3.

hzfanxinxin commented:
@rdblue would you please help to review again, thanks. I've addressed all the comments except one about `HadoopTables` loads the tables which created by `HadoopCatalog`.

rdblue commented on site/docs/api-quickstart.md:
```diff
@@ -48,6 +48,36 @@ logsDF.write
+logsDF.write
+    .format("iceberg")
+    .mode("append")
+    .save("hdfs://warehouse_path/logging/logs")
```
This location also needs to be fixed to avoid using `warehouse_path` as the authority section of the URI.

rdblue commented:
I did some minor editing and fixed the read path. Overall, looks good so I'll merge this.
